sources/packages/backend/apps/queue-consumers/src/processors/virus-scan/virus-scan.processor.ts

```diff
@@ -35,29 +34,23 @@ export class VirusScanProcessor {
       );
       processSummary.info("Completed virus scanning for the file.");
     } catch (error: unknown) {
-      if (error instanceof NotFoundException) {
-        // If the file is not present in the database,
-        // remove the file from the virus scan queue.
-        await job.discard();
-        processSummary.warn(
-          `File ${job.data.uniqueFileName} not found or has already been scanned for viruses. Scanning the file for viruses is aborted.`,
-        );
-      } else if (error instanceof CustomNamedError) {
-        if (error.name === UNABLE_TO_SCAN_FILE) {
-          processSummary.error(
-            `Unable to scan the file ${job.data.uniqueFileName} for viruses.`,
-          );
+      if (error instanceof CustomNamedError) {
+        const errorMessage = error.message;
+        if (error.name === FILE_NOT_FOUND) {
+          // If the file is not present in the database, remove the file from the virus scan queue.
+          await job.discard();
         }
-      } else {
-        processSummary.error(
-          `Error while scanning the file ${job.data.uniqueFileName} for viruses.`,
-        );
+        processSummary.error(errorMessage);
+        throw new Error(errorMessage, { cause: error.name });
       }
```

Comment:
Please throw the error when it is something different from the `CustomNamedError`.
Please add the below at the end of the catch block.
```ts
throw new Error(
  "Unexpected error while executing the job, check logs for further details.",
  { cause: error },
);
```